Lib/subprocess.py

```diff
@@ -1307,15 +1307,15 @@ def _execute_child(self, args, executable, preexec_fn, close_fds,
                 try:
```

Comment:
To simplify the code, I suggest to decode pipe data before the try:

```
errpipe_data = errpipe_data.decode(errors="surrogatepass")
```

Since both code paths decode anyway. It would allow to work on Unicode rather than bytes, which is more convenient.